sys/arduino/SPI.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2019 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     sys_arduino_api
+ * @{
+ *
+ * @file
+ * @brief       Implementation of the Arduino 'SPI' interface
+ *
+ * @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ *
+ * @}
+ */
+
+#ifdef MODULE_PERIPH_SPI
+
+extern "C" {
+#include "assert.h"
+}
+
+#include "SPI.h"
+
+SPISettings::SPISettings(uint32_t clock, uint8_t bitOrder, uint8_t dataMode)
+{
+    static const spi_clk_t clocks[] = {
+        SPI_CLK_10MHZ, SPI_CLK_5MHZ, SPI_CLK_1MHZ, SPI_CLK_400KHZ
+    };
+    static const uint32_t steps [] = {
+        7500000, 3000000, 700000, 250000
+    };
+
+    assert(bitOrder == MSBFIRST);
+    switch(dataMode) {
+        default:
+        case SPI_MODE0:
+            mode = SPI_MODE_0;
+            break;
+        case SPI_MODE1:
+            mode = SPI_MODE_1;
+            break;
+        case SPI_MODE2:
+            mode = SPI_MODE_2;
+            break;
+        case SPI_MODE3:
+            mode = SPI_MODE_3;
+            break;
+    }
+
+    for (uint8_t i = 0; i < ARRAY_SIZE(steps); i++) {
+        if (clock >= steps[i]) {
+            clock = clocks[i];
```

Comment:
Either
```suggestion
            this->clock = clocks[i];
```
or parameter `clock` has to be renamed.